File: Rivington.IG.Web/Rivington.IG.Domain.Models/OrderManagement/InspectionOrderSections/Property.cs
using System;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Newtonsoft.Json;

namespace Rivington.IG.Domain.Models.OrderManagement
{
    public class InspectionOrderProperty
    {
        [JsonIgnore]
        [Parent]
        public InspectionOrder InspectionOrder { get; set; }
        [Key]
        public Guid Id { get; set; }
        public InspectionOrderPropertyGeneral General { get; set; }
        public InspectionOrderPropertyInteriorDetail InteriorDetail { get; set; }
        public InspectionOrderPropertyHomeDetail HomeDetail { get; set; }
        public InspectionOrderPropertyBuildingConcern BuildingConcern { get; set; }
        public InspectionOrderPropertyDetachedStructure DetachedStructure { get; set; }
        public InspectionOrderPropertyAdditionalExposure AdditionalExposure { get; set; }
        public InspectionOrderPropertyHighValueKitchen HighValueKitchen { get; set; }
        public InspectionOrderPropertyHighValueBath HighValueBath { get; set; }
        public InspectionOrderPropertyHighValueFloorToCeiling HighValueFloorToCeiling { get; set; }
        public InspectionOrderPropertyHighValueInteriorFeature HighValueInteriorFeature { get; set; }
        public InspectionOrderPropertyNonWildfireAssessment NonWildfireAssessment { get; set; }

        public static void BuildModel(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<InspectionOrder>()
                .HasOne(a => a.Property).WithOne(b => b.InspectionOrder)
                .HasForeignKey<InspectionOrderProperty>()
                .OnDelete(DeleteBehavior.Cascade);

            InspectionOrderPropertyGeneral.BuildModel(modelBuilder);

            InspectionOrderPropertyInteriorDetail.BuildModel(modelBuilder);

            InspectionOrderPropertyHomeDetail.BuildModel(modelBuilder);

            InspectionOrderPropertyBuildingConcern.BuildModel(modelBuilder);

            InspectionOrderPropertyDetachedStructure.BuildModel(modelBuilder);

            InspectionOrderPropertyAdditionalExposure.BuildModel(modelBuilder);

            InspectionOrderPropertyHighValueKitchen.BuildModel(modelBuilder);

            InspectionOrderPropertyHighValueBath.BuildModel(modelBuilder);

            InspectionOrderPropertyHighValueFloorToCeiling.BuildModel(modelBuilder);

            InspectionOrderPropertyHighValueInteriorFeature.BuildModel(modelBuilder);

            InspectionOrderPropertyNonWildfireAssessment.BuildModel(modelBuilder);
        }
    }
}

File: Rivington.IG.Web/Rivington.IG.API/src/app/modules/shared/models/ordermanagement/inspection-order/checklist/property.ts
import { InspectionOrderPropertyGeneral } from "./property/general";
import { InspectionOrderPropertyInteriorDetail } from "./property/interior-detail";
import { InspectionOrderPropertyHomeDetail } from "./property/home-detail";
import { InspectionOrderPropertyBuildingConcern } from "./property/building-concern";
import { InspectionOrderPropertyDetachedStructure } from "./property/detached-structure";
import { InspectionOrderPropertyAdditionalExposure } from "./property/additional-exposure";
import { InspectionOrderPropertyHighValueKitchen } from "./property/high-value-kitchen";
import { InspectionOrderPropertyHighValueBath } from "./property/high-value-bath";
import { InspectionOrderPropertyHighValueFloorToCeiling } from "./property/high-value-floor-to-ceiling";
import { InspectionOrderPropertyHighValueInteriorFeature } from "./property/high-value-interior-feature";
import { InspectionOrderPropertyNonWildfireAssessment } from "./mitigation/InspectionOrderNonWildfireAssessment";

export class InspectionOrderProperty {
    id?: string;
    general?: InspectionOrderPropertyGeneral;
    interiorDetail?: InspectionOrderPropertyInteriorDetail;
    homeDetail?: InspectionOrderPropertyHomeDetail;
    buildingConcern?: InspectionOrderPropertyBuildingConcern;
    detachedStructure?: InspectionOrderPropertyDetachedStructure;
    additionalExposure?: InspectionOrderPropertyAdditionalExposure;
    highValueKitchen?: InspectionOrderPropertyHighValueKitchen;
    highValueBath?: InspectionOrderPropertyHighValueBath;
    highValueFloorToCeiling?: InspectionOrderPropertyHighValueFloorToCeiling;
    highValueInteriorFeature?: InspectionOrderPropertyHighValueInteriorFeature;
    nonWildfireAssessment?: InspectionOrderPropertyNonWildfireAssessment;
    
    public constructor(init?:Partial<InspectionOrderProperty>) {
        Object.assign(this, init);
    }
}